apps/designer/app/canvas/shared/instance.ts

```diff
@@ -48,14 +51,40 @@ export const usePopulateRootInstance = (tree: Tree) => {
   }
 };
 
+const findInsertLocation = (
```

Comment:
we could potentially have test for this? or is this tested through something else/